hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/snapshot/OMSnapshotPurgeResponse.java

```diff
@@ -96,8 +97,9 @@ protected void addToDBBatch(OMMetadataManager omMetadataManager,
 
       // Delete Snapshot checkpoint directory.
       deleteCheckpointDirectory(omMetadataManager, snapshotInfo);
-      omMetadataManager.getSnapshotInfoTable().deleteWithBatch(batchOperation,
-          dbKey);
+      omMetadataManager.getSnapshotInfoTable().deleteWithBatch(batchOperation, dbKey);
+      ((OmMetadataManagerImpl) omMetadataManager).getSnapshotChainManager()
+          .deleteSnapshotFromSnapshotIdToTableKey(snapshotInfo.getSnapshotId());
```

Comment:
I guess we should be removing the snapshot from the snapshot chain as soon as we remove from the cache. This operation I believe needs to go in OMSnapshotPurgeRequest.

Reply:
The validateAndUpdateCache method should be removing from snapshot chain

Reply:
First, this is not removing from the chain, it is just removing from the reverse mapping of snapshotId to snapshotTable key we maintain. 
Second, you are not completely wrong and it was like this before. But I saw some unnecessary error logs and  BackGoundService failures because till the double buffer gets flushed BackGoundService which iterates over the table can see the purged snapshot (not flushed snapshot) but when the service tries to load it, it falls because reverse mapping is not present.
Hence I extracted out the delete from the `snapshotIdToTableKey` map as a separate function.

Reply:
We can see if we can remove this after https://github.com/apache/ozone/pull/6296.

Reply:
Reason for the changes, after the change I was seeing an error in [SnapshotDeletingService](https://github.com/apache/ozone/blob/2f2234c7b61714404399ada8f31b3fb4772b613a/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java#L167) if doubleBuffer doesn't get flush by the time next run of the SnapshotDeletingService starts.

Had an offline discussion and it will fail eventually either here or later on the purge path. Reverting as suggested.

Thanks @swamirishi for the comment.